### PR TITLE
Check commits

### PR DESCRIPTION
**Pull request by taquitos**
- rename some methods to be more obvious that they are asynchronous
- be explicit about storing build timestamps as utc
- create_and_queue_build_task(sha:) now creates a new repo each time

### CODE REVIEW

File: app/workers/check_for_new_commits_on_github_worker.rb [hunk 1]
```diff
@@ -14,32 +16,49 @@ class CheckForNewCommitsOnGithubWorker < GitHubWorkerBase
...
 
       super(provider_credential: provider_credential, project: project) # This starts the work by calling `work`
     end
 
-    def work
-      logger.debug("Checking for new commits in #{self.project.project_name}")
-      repo = self.git_repo
-
-      # TODO: ensure BuildService subclasses are thread-safe
+    def check_for_new_commits
```
Comment (taquitos) [inline app/workers/check_for_new_commits_on_github_worker.rb]: this is much cleaner if you look at the whole file instead of this diff. It's very understandable

File: app/workers/github_worker_base.rb [hunk 2]
```diff
@@ -88,11 +88,19 @@ def target_branches(&block)
...
     end
 
-    def create_and_queue_build_task(sha:, repo:)
+    def create_and_queue_build_task(sha:)
       credential = self.provider_credential
       current_project = self.project
       current_sha = sha
       return unless Services.build_runner_service.find_build_runner(project_id: current_project.id, sha: current_sha).nil?
+
+      repo = GitRepo.new(
```
Comment (taquitos) [inline app/workers/github_worker_base.rb]: new clone for each sha 🎉

File: app/features/build_runner/build_runner.rb [hunk 3]
```diff
@@ -214,7 +214,7 @@ def prepare_build_object
         project: self.project,
         number: new_build_number,
         status: :pending,
-        timestamp: Time.now,
+        timestamp: Time.now.utc,
```
Comment (KrauseFx) [inline app/features/build_runner/build_runner.rb]: Could you add a comment to the code here on why this change is necessary? I assume this depends on the server's time zone?

File: app/workers/check_for_new_commits_on_github_worker.rb [hunk 4]
```diff
@@ -4,6 +4,8 @@
 require_relative "../shared/models/job_trigger"
 require_relative "../shared/logging_module"
 
+require "time"
```
Comment (KrauseFx) [inline app/workers/check_for_new_commits_on_github_worker.rb]: Sounds like this requires some time 

🦆s after adding this 👨 joke

File: app/workers/check_for_new_commits_on_github_worker.rb [hunk 5]
```diff
@@ -14,32 +16,54 @@ class CheckForNewCommitsOnGithubWorker < GitHubWorkerBase
...
+      sorted_builds = build_service.list_builds(project: self.project).sort { |x, y| y.timestamp.to_i <=> x.timestamp.to_i }
 
-        # Skips branches that have previously been built
-        builds = build_service.list_builds(project: self.project)
-        next if builds.map(&:sha).include?(current_sha)
+      # All the shas for builds we have run, sorted by build timestamp (newest builds first)
+      local_build_shas_by_date = sorted_builds.map(&:sha)
 
-        logger.debug("Detected new commit in #{self.project.project_name} on branch #{branch.name} with sha #{current_sha}")
+      # Look at the last commit time, then subtract 5 minutes from it to account for clock drift between this computer and the git repo
```
Comment (KrauseFx) [inline app/workers/check_for_new_commits_on_github_worker.rb]: Great comments

File: app/features/build_runner/build_runner.rb (hunk 3 above)
Comment (taquitos) [reply app/features/build_runner/build_runner.rb]: exactly.

File: app/workers/check_for_new_commits_on_github_worker.rb (hunk 4 above)
Comment (taquitos) [reply app/workers/check_for_new_commits_on_github_worker.rb]: shockedguy
